### PR TITLE
Fix resource registration for native compilation

### PR DESCRIPTION
**Pull request by zakkak**
NativeImageResourceBuildItem and ServiceProviderBuildItem contain a path
of the resource we should include and not a pattern or a glob. As a
result, `Pattern.quote` is the right method to use in order to produce a
pattern that would match the path.

The patch also remove the wrong addition of the paths "as is" to the
includes json array.

Fix up of b7f49dd9d952a46fde3f0abaabcc08b1a2e5f067 (https://github.com/quarkusio/quarkus/pull/31185)

### CODE REVIEW

**Review by gastaldi — APPROVED** [review]

**Review by dmlloyd — APPROVED** [review]

**Review by mkouba — COMMENTED** [review]
I believe that we should also clarify the javadoc of `NativeImageResourceBuildItem` so that it's clear what exactly does `resource`  mean there and how it's handled later on in the `NativeImageResourceConfigStep`...

**Comment by zakkak** [comment]
> I believe that we should also clarify the javadoc of `NativeImageResourceBuildItem` so that it's clear what exactly does `resource` mean there and how it's handled later on in the `NativeImageResourceConfigStep`...

Done in https://github.com/quarkusio/quarkus/pull/40184/commits/84f3653c3f26379a6b7a0d5e21f41bdcf8a85e9e

**Review by gsmet — APPROVED** [review]
Makes sense for `main` and `3.10`.
I will send an email to quarkus-dev to check it won't break people before backporting further.

File: core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/NativeImageResourceBuildItem.java [hunk 1]
```diff
@@ -7,7 +7,11 @@
 import io.quarkus.builder.item.MultiBuildItem;
 
 /**
- * A build item that indicates that a static resource should be included in the native image
+ * A build item that indicates that a static resource should be included in the native image.
+ * <p>
+ * A static resource is a file that is not processed by the build steps, but is included in the native image as-is.
```
Comment (mkouba) [inline core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/NativeImageResourceBuildItem.java:12]: I thought we should mention that the resource is a `/`-separated path name (with the same semantics as `java.lang.ClassLoader.getResource(String)`?).

Comment (zakkak) [reply core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/NativeImageResourceBuildItem.java:12]: Ack, please take another look.

**Review by mkouba — APPROVED** [review]

**Pull request merged**